docs/guides/flex-cluster-to-dedicated-cluster-migraton-guide.md

```diff
@@ -0,0 +1,46 @@
+---
+page_title: "Migration Guide: FLex Cluster to Dedicated Cluster"
+---
+
+# Migration Guide: Flex Cluster to Dedicated Cluster
+
+**Objective**: This guide explains how to replace the `mongodbatlas_flex_cluster` resource with the `mongodbatlas_advanced_cluster` resource.
+
+Currently, the only method to migrate your Flex cluster to a Dedicated cluster is via the Atlas UI.
+
+<!-- Noting that implementation of flex_cluster as a part of mongodb_advanced_cluster in January 2025 will create new migration journey -->
+
+## Best Practices Before Migrating
+Before doing any migration, create a backup of your [Terraform state file](https://developer.hashicorp.com/terraform/cli/commands/state).
+
+### Procedure
+
+
+See [Modify a Cluster](https://www.mongodb.com/docs/atlas/scale-cluster/) for how to migrate via the Atlas UI.
+
+The following resolves the configuration drift in Terraform and will not affect the underlying cluster infrastructure:
+
+1. Find the import IDs of the new Dedicated cluster your Flex cluster has migrated to: `{PROJECT_ID}-{CLUSTER_NAME}`, such as `664619d870c247237f4b86a6-clusterName`
+2. Add an import block to on of your `.tf` files:
+  ```terraform
+  import {
+    to = mongodbatlas_advanced_cluster.this
+    id = "664619d870c247237f4b86a6-clusterName" # from step 1
+  }
+  ```
+  3. Run `terraform plan -generate-config-out=adv_cluster.tf`. This should generate a `adv_cluster.tf` file.
```

Comment:
IMO this approach looks "overkill" 😅
I think the schema of `mongodbatlas_flex_cluster` is simpler and this guide could instead describe manual changes instead of generating a full new config.

Reply:
I feel like a guide to manually change `mongodbatlas_flex_cluster` to a `mongodbatlas_advanced_cluster` is just as complicated as generating and adjusting a new config. The only difference is steps 5&6 would outline how to change the resource block and add necessary attributes. 

At least generating a new config will ensure that the user has all the correct attributes instead of entering some attribute incorrectly and chancing changing their infrastructure. WDYT?

I will change steps 5&6 to more clearly outline what needs to be adjusted from the imported dedicated cluster.

Reply:
No worries, feel free to keep it 👍